cfme/utils/log_validator.py

```diff
@@ -49,32 +60,46 @@ def __init__(self, remote_filename, **kwargs):
         self._remote_file_tail = SSHTail(remote_filename, **kwargs)
         self._matches = {key: 0 for key in self.matched_patterns}
 
-    def fix_before_start(self):
+    def start_monitoring(self):
         """Start monitoring log before action"""
         self._remote_file_tail.set_initial_file_end()
+        logger.info("Log monitoring has been started on remote file")
 
     def _check_skip_logs(self, line):
         for pattern in self.skip_patterns:
             if re.search(pattern, line):
-                logger.info('Skip pattern {} was matched on line {},\
-                            so skipping this line'.format(pattern, line))
+                logger.info(
+                    "Skip pattern %s was matched on line %s so skipping this line", pattern, line
+                )
                 return True
         return False
 
     def _check_fail_logs(self, line):
         for pattern in self.failure_patterns:
             if re.search(pattern, line):
-                pytest.fail('Failure pattern {} was matched on line {}'.format(pattern, line))
+                logger.error("Failure pattern %s was matched on line %s", pattern, line)
+                raise LogValidatorException(pattern, "Expected failure pattern found in log.", line)
 
     def _check_match_logs(self, line):
         for pattern in self.matched_patterns:
             if re.search(pattern, line):
-                logger.info('Expected pattern {} was matched on line {}'.format(pattern, line))
+                logger.info("Expected pattern %s was matched on line %s", pattern, line)
                 self._matches[pattern] = self._matches[pattern] + 1
 
+    @property
+    def _is_valid(self):
+        for pattern, count in self.matches.items():
+            if count == 0:
+                logger.info("Expected '%s' pattern not found", pattern)
+                return False
+        return True
+
     @property
     def matches(self):
-        """It will return pattern match count dictionary"""
+        """Collect match count in log
+
+        Returns (dict): Pattern match count dictionary
```

Comment:
_Optional_ thoughts on also giving example of the `key`, and `values` of the dict? I.e. Key is the `match` and value is the `count`?